### ISSUE TITLE
HDSpain add Megapack label to title

### ISSUE THREAD

**Issue by R91g** [Jackett/Jackett#5326]
Hi,

In HDSpain when a Season is finish they upload all episodies in one Megapack

This information is save in a label "**td.titulo a[class]**"  but I don't know how add this to the title selector:

>       title:
>         selector: td.titulo a[id]
>         filters:
>           - name: append
>             args: " [spanish]"

**Comment by R91g** [comment]
Partially fixed with #5327

**Comment by garfield69** [comment]
>In HDSpain when a Season is finish they upload all episodies in one Megapack
>This information is save in a label "td.titulo a[class]" but I don't know how add this to the title selector:

I have not seen the HTML produced by HDSpain, so here is an untested example of extracting optional data and appending it to title from which you can draw inspiration:
```yaml
  megapack:
    optional: true
    selector: td.titulo a[class]:contains("Megapack")
  title:
    selector: td.titulo a[id]
    filters:
      - name: append
        args: " [spanish]{{.if .Result.megapack}} {{.Result.megapack}}{{else}}{{end}} "
```
the `{{ .Result.$FieldName }}` variable can be used for any fields you have in your row block. If the variable always exists you can drop the optional statement, the contains-test and the if.else.end string expression.

more at https://github.com/Jackett/Jackett/wiki/Definition-format#variables-available-during-search-queries,  https://github.com/Jackett/Jackett/wiki/Definition-format#template-engine and https://github.com/Jackett/Jackett/wiki/Definition-format#variables

**Comment by R91g** [comment]
Thank you this example is very inspiration.

Now the problem is If the first result contains "Megapack" in the rest it also adds it even if it is not a megapack

I need to empty the megapack variable after each result

**Comment by garfield69** [comment]
perhaps adding this before the current megapack
```yaml
  megapack:
    text: ""
```

**Comment by garfield69** [comment]
then the title can just be reduced to
```yaml
      - name: append
        args: " [spanish] {{.Result.megapack}} "
```

**Comment by R91g** [comment]
Now works perfect!
Thank you.

I will close the pull #5327 and do a new with this changes

**Comment by garfield69** [comment]
no need to close the pull. simple edit your existing source and it will be included automatically.

**Comment by R91g** [comment]
Sorry I closed the pull and do a new #5330
